src/ripple/nodestore/ShardSizeTuning.md

```diff
@@ -141,6 +141,32 @@ sizes of nudb.* which are permanent for deterministic shards.
 We can see that in all tables ratio is greater then 1, so using shards with
 16 ledgers is preferred.
 
+## Compare 16K shards and 32K shards
+
+To claim that shards with 16K ledgers are the best choise, we also assembled
+shards with 32 ledgers per shard. There were 11 such a shards computed with
+indexes from 1300 to 1800 with step 50. These are the shards which correspond
+to previously calculated shards with 16 ledgers.
+
+Here are results for these shards. Average size of nudb.dat is 6.74G, average
+sum of sizes for nudb.dat and nudb.key is 7.65G. Now we can compare these sizes
+and doubled average sizes for 16k-ledgers shards. We have
+
+3.89 * 2 = 7.78 greather than 6.74 by 15 percents.
+
+4.43 * 2 = 8.86 greather than 7.65 by 16 percents.
+
 ## Conclusion
 
-We recommend use shards with 16K ledgers.
+We showed that using shards with 8k ledgers leads to raising required disk
+size by 20% in comparison with using shards with 16k ledgers. In the same way,
+using shards with 16k ledgers leads to raising required disk space by 15%
+in comparison with using shards with 16k ledgers. So if we only take into
+consideration disk usage, then optimal is using shards with 32k ledgers.
+But we also should take into consideration computational power of nodes.
+For example, my machine with 32G memory is best for acquire shards with 16k
```

Comment:
Really? A 32k shard fits 4 times into 32GiB of memory. What makes this "best" for half the size?

Reply:
There are also some internal structures in rippled which occupy the memory.

Reply:
Yes? Still `rippled` uses nowhere near this amount of memory on my node and if it really keeps several copies of the same data in memory when dealing with or constructing shards, this seems like a serious issue.

Reply:
Ok, I will check 32k ledgers shards on my 32G laptop. Generally said, Ripple/XRP is serious issue itself.

Reply:
Checked on 32k ledgers shards. First shard successfully acquired, and the program stopped downloading shards at this point, none further shards downloading. May be compound reason: lack of resources + software bug.